commons/src/main/java/com/powsybl/commons/datasource/ReadOnlyMemDataSource.java

```diff
@@ -63,24 +64,27 @@ public String getBaseName() {
     }
 
     @Override
-    public boolean exists(String suffix, String ext) throws IOException {
-        return exists(DataSourceUtil.getFileName(baseName, suffix, ext));
+    public boolean exists(String suffix, String ext, boolean checkConsistencyWithDataSource) throws IOException {
+        return exists(DataSourceUtil.getFileName(baseName, suffix, ext), checkConsistencyWithDataSource);
     }
 
     @Override
-    public boolean exists(String fileName) throws IOException {
+    public boolean exists(String fileName, boolean checkConsistencyWithDataSource) throws IOException {
         Objects.requireNonNull(fileName);
-        return data.containsKey(fileName);
+        return (!checkConsistencyWithDataSource || isConsistentWithDataSource(fileName)) && data.containsKey(fileName);
     }
 
     @Override
-    public InputStream newInputStream(String suffix, String ext) throws IOException {
+    public InputStream newInputStream(String suffix, String ext, boolean checkConsistencyWithDataSource) throws IOException {
         return newInputStream(DataSourceUtil.getFileName(baseName, suffix, ext));
```

Comment:
should pass checkConsistencyWithDataSource ?

Reply:
fixed